### PR TITLE
Tries to guess which qemu executable should use via WORKER_CLASS

### PR DESCRIPTION
**Pull request by dasantiago**
related poo#14674

There are two ways to define which qemu to use:
- via environment variable QEMU
- via setting the variable QEMU (it will be appended to qemu-system-)

However some templates don't have this variable defined, inducing
the worker to choose the wrong qemu.
So if any of the previous mechanisms work, it will use the
WORKER_CLASS to guess which executable should use.

### CODE REVIEW

File: backend/qemu.pm [hunk 1]
```diff
@@ -418,8 +418,15 @@ sub start_qemu {
...
 
     my $qemuimg = find_bin('/usr/bin/', qw(kvm-img qemu-img));
+
     unless ($qemubin) {
-        $qemubin = find_bin('/usr/bin/', $vars->{QEMU} ? ('qemu-system-' . $vars->{QEMU}) : qw(kvm qemu-kvm qemu qemu-system-x86_64 qemu-system-ppc64));
+        if ($vars->{QEMU}) {
+            $qemubin = find_bin('/usr/bin/', 'qemu-system-' . $vars->{QEMU});
+        }
+        else {
+            (my $class = $vars->{WORKER_CLASS}) =~ s/qemu_/qemu-system\-/g;
```
Comment (coolo) [inline backend/qemu.pm]: won't this cause warnings on undefined?

Comment (dasantiago) [reply backend/qemu.pm]: @coolo 

Yes it will. :-)

Hence the 
> (split(/\s*,\s*/, $class // '')

in the next line

Comment (dasantiago) [reply backend/qemu.pm]: @coolo true.

Fixed.

File: backend/qemu.pm [hunk 2]
```diff
@@ -418,8 +418,15 @@ sub start_qemu {
...
     my $qemuimg = find_bin('/usr/bin/', qw(kvm-img qemu-img));
+
     unless ($qemubin) {
-        $qemubin = find_bin('/usr/bin/', $vars->{QEMU} ? ('qemu-system-' . $vars->{QEMU}) : qw(kvm qemu-kvm qemu qemu-system-x86_64 qemu-system-ppc64));
+        if ($vars->{QEMU}) {
+            $qemubin = find_bin('/usr/bin/', 'qemu-system-' . $vars->{QEMU});
+        }
+        else {
+            (my $class = $vars->{WORKER_CLASS} || '') =~ s/qemu_/qemu-system\-/g;
+            $qemubin = find_bin('/usr/bin/', (split(/\s*,\s*/, $class), qw(kvm qemu-kvm qemu qemu-system-x86_64 qemu-system-ppc64)));
```
Comment (coolo) [inline backend/qemu.pm]: I have one more question: if WORKER_CLASS=tap,qemu_x86_64 will you start /usr/bin/tap?

Comment (dasantiago) [reply backend/qemu.pm]: @coolo You do good questions :+1:  :-D

got the point.

Fixed

File: backend/qemu.pm [hunk 3]
```diff
@@ -418,8 +418,19 @@ sub start_qemu {
...
+        if ($vars->{QEMU}) {
+            $qemubin = find_bin('/usr/bin/', 'qemu-system-' . $vars->{QEMU});
+        }
+        else {
+            (my $class = $vars->{WORKER_CLASS} || '') =~ s/qemu_/qemu-system\-/g;
+            my %allowed = map { $_ => 1 } qw(kvm qemu-kvm qemu qemu-system-x86_64 qemu-system-ppc64);
+            for (split(/\s*,\s*/, $class)) {
+                $qemubin = find_bin('/usr/bin/', $_) if $allowed{$_};
+            }
+            $qemubin = find_bin('/usr/bin/', keys %allowed) unless $qemubin;
```
Comment (coolo) [inline backend/qemu.pm]: this drops order - and we would run qemu-system-ppc64 before qemu by chance (perl hash randomization)

Comment (dasantiago) [reply backend/qemu.pm]: @coolo done.

**Comment by coolo** [comment]
you thought it's an easy one, didn't you? :)

**Pull request merged**